docs/proposals/device-crd.md

```diff
@@ -0,0 +1,532 @@
+|Device Deleted                 | The controller sends the device twin delete event to delete all device twins associated with the device. It also deletes config maps associated with the device and this delete event is synced to the edge. The mapper application effectively stops operating on the device.|
+|Device Model Deleted           |  The controller needs to run [`finalizers`](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) to ensure that all device instances using the device model are deleted first, and only then should the model deletion proceed.|
+
+The idea behind using config map to store device properties and visitors is that these metadata are only required by the mapper applications running on the edge node in order to connect to the device and collect data.
+Mappers if run as containers can load these properties as config maps . Any additions , deletions or updates to properties , visitors etc in the cloud are watched upon by the downstream controller and config maps are updated in etcd. The existing edge controller already has the mechanism to watch on config map updates and push them to the edge node. A mapper application can get these updates and then adjust the data collection process.
+A separate design proposal can be prepared to illustrate the details of how mappers can leverage these config maps.
+
+A sample config map for the device model described earlier is shown below
+
+### Device Config Map sample
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: device1-config
+  namespace: foo
+data:
+  deviceProfile.json: |-
+    {
+      "device_instances": [
+        {
+          "id": "1",
+          "name": "device1",
+          "protocol": "BluetoothLE",
+          "model": "SensorTagModel"
+        }
+      ],
+      "device_models": [
+        {
+          "name": "SensorTagModel",
+          "description": "TI Simplelink SensorTag Device Attributes Model",
+          "properties": [
+            {
+              "name": "temperature",
+              "datatype": "float",
+              "accessMode": "r",
+              "unit": "degree fahrenheit"
+            },
+            {
+              "name": "temperatureEnable",
+              "datatype": "float",
+              "accessMode": "r",
+              "unit": "degree fahrenheit"
+            }
+          ]
+        }
+      ],
+      "protocols": [
+        {
+          "name": "BluetoothLE",
+          "protocolConfig": {}
+        }
+      ],
+      "property_visitors": [
+        {
+          "name": "temperature",
+          "model_name": "SensorTagModel",
+          "protocol": "BluetoothLE",
+          "metadata": {
+            "uuid": "F000AA01-0451-4000-B000-000000000000"
+          }
+        },
+        {
+          "name": "temperatureEnable",
```

Comment:
as the same we use name as indentifer.  Add property_name to show which property in the model.

Reply:
added propertyName. please check.